File: IT/1/Variabili, operatori, strutture di controllo/Esercizi/C++/esercizio25.cpp
/* 
file esercizio25.cpp
compilare con: g++ -std=c++11 esercizio25.cpp
*/

#include <iostream>

using std::cin;
using std::cout;
using std::endl;
using std::string;

int main() 
{
	cout << "Scrivi un numero: ";

	string s;
	getline(cin, s);

	if (s.find_first_not_of("0123456789.") != string::npos
		|| s.find_first_of(".") == s.size()-1) {
		cout << "Numero non valido!" << endl;
		return 0;
	}

	double x = stod(s);

	double xv = x >= 0 ? x : -x;

	cout << "Valore assoluto di " << x << " = " << xv << endl;
}

File: IT/1/Variabili, operatori, strutture di controllo/Esercizi/Java/Esercizio25.java
/* file Esercizio25.java */

public class Esercizio25 {
	public static void main(String [] args) {
		String s = System.console().readLine("Scrivi un numero: ");

		// se non è un numero intero oppure in virgola mobile
		if (!s.matches("[\\d]+(\\.[\\d])?")) {
			System.out.println("Numero non valido!");
			return;
		}

		double x = Double.parseDouble(s);

		double xv = x >= 0 ? x : -x;

		System.out.println("Valore assoluto di " + x + " = " + xv);
	}
}

